modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/schema/builder/SchemaBuilder.java

```diff
@@ -722,6 +735,11 @@ ExtensionType registerExecutableType(String name, ParametrizedModel parametrized
             }
         }
 
+        if (base.equals(MULE_ABSTRACT_MESSAGE_SOURCE_TYPE))
```

Comment:
"tell, don't ask". Accumulating ifs like this doesn't scale well. I really liked the maneuver on ConnectionProviderSchemaDelegate in which you tell the builder to add a retry policy. Do the same here, and instead of asking for this condition, have the SourceSchemaDelegate tell the builder to add the retry policy